tool/tctl/common/auth_command.go

```diff
@@ -469,92 +467,54 @@ func (a *AuthCommand) generateDatabaseKeys(ctx context.Context, clusterAPI auth.
+	filesWritten, err := db.GenerateDatabaseCertificates(ctx, genMTLSReq)
 	if err != nil {
 		return trace.Wrap(err)
 	}
-	resp, err := clusterAPI.GenerateDatabaseCert(ctx,
-		&proto.DatabaseCertRequest{
-			CSR: csr,
-			// Important to include SANs since CommonName has been deprecated
-			// since Go 1.15:
-			//   https://golang.org/doc/go1.15#commonname
-			ServerNames: principals,
-			// Include legacy ServerName for compatibility.
-			ServerName:    principals[0],
-			TTL:           proto.Duration(a.genTTL),
-			RequesterName: proto.DatabaseCertRequest_TCTL,
-		})
-	if err != nil {
-		return trace.Wrap(err)
+
+	return trace.Wrap(writeHelperMessageDBmTLS(os.Stdout, filesWritten, a.output, a.outputFormat))
+}
+
+var mapIdentityFileFormatHelperTemplate = map[identityfile.Format]*template.Template{
+	identityfile.FormatDatabase:  dbAuthSignTpl,
+	identityfile.FormatMongo:     mongoAuthSignTpl,
+	identityfile.FormatCockroach: cockroachAuthSignTpl,
+	identityfile.FormatRedis:     redisAuthSignTpl,
+	identityfile.FormatSnowflake: snowflakeAuthSignTpl,
+}
+
+func writeHelperMessageDBmTLS(writer io.Writer, filesWritten []string, output string, outputFormat identityfile.Format) error {
+	if writer == nil {
+		return nil
 	}
-	key.TLSCert = resp.Cert
-	key.TrustedCA = []auth.TrustedCerts{{TLSCertificates: resp.CACerts}}
-	filesWritten, err := identityfile.Write(identityfile.WriteConfig{
-		OutputPath:           a.output,
-		Key:                  key,
-		Format:               a.outputFormat,
-		OverwriteDestination: a.signOverwrite,
-	})
-	if err != nil {
-		return trace.Wrap(err)
+
+	tpl, found := mapIdentityFileFormatHelperTemplate[outputFormat]
+	if !found {
+		// This format doesn't have a recommended configuration.
+		// Consider adding one to ease the installation for the end-user
+		return nil
 	}
-	switch a.outputFormat {
-	case identityfile.FormatDatabase:
-		err = dbAuthSignTpl.Execute(os.Stdout, map[string]interface{}{
-			"files":  strings.Join(filesWritten, ", "),
-			"output": a.output,
-		})
-	case identityfile.FormatMongo:
-		err = mongoAuthSignTpl.Execute(os.Stdout, map[string]interface{}{
-			"files":  strings.Join(filesWritten, ", "),
-			"output": a.output,
-		})
-	case identityfile.FormatCockroach:
-		err = cockroachAuthSignTpl.Execute(os.Stdout, map[string]interface{}{
-			"files":  strings.Join(filesWritten, ", "),
-			"output": a.output,
-		})
-	case identityfile.FormatRedis:
-		err = redisAuthSignTpl.Execute(os.Stdout, map[string]interface{}{
-			"files":  strings.Join(filesWritten, ", "),
-			"output": a.output,
-		})
-	case identityfile.FormatSnowflake:
-		err = snowflakeAuthSignTpl.Execute(os.Stdout, map[string]interface{}{
-			"files": strings.Join(filesWritten, ", "),
-		})
+
+	tplVars := map[string]interface{}{
+		"files":  strings.Join(filesWritten, ", "),
+		"output": output,
 	}
-	return trace.Wrap(err)
+
+	if outputFormat == identityfile.FormatSnowflake {
+		delete(tplVars, "output")
+	}
```

Comment:
Do we need this? If the template variable is present but just not used in the template, it should be fine, right?